app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -115,54 +115,61 @@ def update_auth_header
       response.headers.merge!(auth_header)
 
     else
-
-      ensure_pristine_resource do
-        # Lock the user record during any auth_header updates to ensure
-        # we don't have write contention from multiple threads
-        @resource.with_lock do
-          # should not append auth header if @resource related token was
-          # cleared by sign out in the meantime
-          return if @used_auth_by_token && @resource.tokens[@client_id].nil?
-
-          # determine batch request status after request processing, in case
-          # another processes has updated it during that processing
-          @is_batch_request = is_batch_request?(@resource, @client_id)
-
-          auth_header = {}
-
-          # extend expiration of batch buffer to account for the duration of
-          # this request
-          if @is_batch_request
-            auth_header = @resource.extend_batch_buffer(@token, @client_id)
-
-            # Do not return token for batch requests to avoid invalidated
-            # tokens returned to the client in case of race conditions.
-            # Use a blank string for the header to still be present and
-            # being passed in a XHR response in case of
-            # 304 Not Modified responses.
-            auth_header[DeviseTokenAuth.headers_names[:"access-token"]] = ' '
-            auth_header[DeviseTokenAuth.headers_names[:"expiry"]] = ' '
-
-          # update Authorization response header with new token
-          else
-            auth_header = @resource.create_new_auth_token(@client_id)
-          end
-
-          # update the response header
-          response.headers.merge!(auth_header)
-
-        end # end lock
-      end # end ensure_pristine_resource
+      unless @resource.valid?
+        @resource = resource_class.find(@resource.id) # errors are keep after reload
```

Comment:
Why don't you use `find_resource` method from `ResourceFinder`?

Reply:
I am not looking for email, just reloading the `@resource`, it should be done this way because errors aren't gone after `@resource.reload`, I know it's kind of nasty solution but I can't find something better, check the comment below in order to see why I'm looking for errors.

Reply:
`find_resource` is suitable is not only for email. Yes, `find_resource` is unnecessarily complex. I have already fix `find_resource` in `mongoid` pr and the method works with different ORMs.
Use `@resource.to_param` instead of `@resource.id`. https://api.rubyonrails.org/classes/ActiveModel/Conversion.html#method-i-to_param

Reply:
doesn't Mongo have id? I will use `to_param`, thanks

Reply:
`id` is custom field. `_id` is unique identifier of document in collection and it is storing as BSON. So, you should use `user._id.to_s` or `user.to_param` to get string value of unique id of a document.